File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height) - 1
        max_water = 0

        while l < r:
            h = min(height[l], height[r])
            m = r - l
            max_water = max(max_water, h * m)

            if height[l] < height[r]:
                l+=1
            else:
                r-=1
        return max_water

File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let l = 0
    let r = height.length - 1
    let vol = 0
    while(l < r){
        let minim = Math.min(height[l] , height[r])
        let selisih = r - l
        let res = minim * selisih

        vol = Math.max(vol, res)
        if(height[l] < height[r]){
            l++
        }else if(height[l] > height[r]){
            r--
        }
        else{
            l++
        }
    }

    return vol
};